Deploy build status tidy (#341)
compose-spec/compose-spec: build.md

# The Compose Specification - Build support
{:.no_toc}

*Note:* Build is an OPTIONAL part of the Compose specification.

* ToC
{:toc}

## Introduction

> The Compose specification is a platform-neutral way to define multi-container applications. A Compose implementation which focuses on running an application on a local machine will obviously need to alo support (re)building

Reviewer: ```suggestion
The Compose specification is a platform-neutral way to define multi-container applications. A Compose implementation which focuses on running an application on a local machine will obviously need to also support (re)building
```

application from source. The Compose Build specification lets developers define the build process within a Compose file
in a portable way.

## Definitions

Compose Specification can be extended to support an OPTIONAL `build` subsection on services. 

> This section defines the build requirements for a service container image. Only a subset of Compose file services MAY define such a `build`

Reviewer: This sort of reads like not ALL services in a Compose file MAY define a build section, while what it's trying to say is that not all services MUST define a build section as far as I can tell. Maybe we can reword this?

subsection, others may make use of the `image` attribute. When a `build` subsection is present for a service, it
is valid for Compose to miss out the `image` attribute for the corresponding service, as the Compose implementation
can build an image from source.

Build can be either specified as a single string defining a context path, or as a detailed build definition.

In the former case, the whole path is used as a Docker context to execute a Docker build, looking for a canonical
`Dockerfile` at the root of the directory. The path can be absolute or relative. If it is relative, it MUST be resolved

> from the Compose file parent folder. If it is absolute, the path prevents the Compose file from being portable so the Compose implementation SHOULD warn user accordingly.

Reviewer: For consistency with previous wording changes, maybe use
```suggestion
from the Compose file parent folder. If it is absolute, the path prevents the Compose file from being portable so the Compose implementation SHOULD warn users accordingly.
```

In the later case, build arguments can be specified, including an alternate `Dockerfile` location. The path can be absolute or relative. If it is relative, it MUST be resolved

> from the Compose file parent folder. If it is absolute, the path prevents the Compose file from being portable so the Compose implementation SHOULD warn user accordingly.

Reviewer: same here:
```suggestion
from the Compose file parent folder. If it is absolute, the path prevents the Compose file from being portable so the Compose implementation SHOULD warn user saccordingly.
```

## Consistency with `image`

When a service definition includes both the `image` attribute and a `build` section, the Compose implementation can't
guarantee a pulled image is strictly equivalent to building the same image from source. Without any explicit
user directives, the Compose implementation with Build support MUST first try to pull the image, then build from source
if the image was not found on registry. The Compose implementation MAY offer options to customize this behaviour by user
request.

## Publishing built images

The Compose implementation with Build support SHOULD offer an option to push built images to a registry. When doing so, it

> MUST NOT try to push service images without an `image` attribute. The Compose implementation SHOULD warn user about the missing `Image` attribute which prevent images being pushed.

Reviewer: ```suggestion
MUST NOT try to push service images without an `image` attribute. The Compose implementation SHOULD warn users about the missing `Image` attribute which prevent images being pushed.
```

### dockerfile

> `dockerfile` lets you set an alternate Dockerfile. A relative path MUST be resolved from the build context.

Reviewer: ```suggestion
`dockerfile` lets users set an alternate Dockerfile. A relative path MUST be resolved from the build context.
```

Reviewer: or alternatively
```suggestion
`dockerfile` sets an alternate Dockerfile. A relative path MUST be resolved from the build context.
```

### dockerfile_inline

> `dockerfile_inline` lets you define Dockerfile content as inlined string in a Compose file. When set, the `dockerfile` 

Reviewer: ```suggestion
`dockerfile_inline` lets users define Dockerfile content as inlined string in a Compose file. When set, the `dockerfile` 
```

Reviewer: or 
```suggestion
`dockerfile_inline` defines Dockerfile content as inlined string in a Compose file. When set, the `dockerfile` 
```
